packages/web-app-files/src/components/SideBar/Shares/FileLinks.vue

```diff
@@ -261,14 +324,133 @@ export default defineComponent({
+          .setLocale(this.$language.current)
+          .endOf('day')
+          .toFormat("yyyy-MM-dd'T'HH:mm:ssZZZ")
+      }
+
+      let password
+
+      switch (link.password) {
+        // no password
+        case false:
+          password = undefined
+          break
+        // delete password
+        case '':
+          password = ''
+          break
+        // existing password
+        case true:
+          password = undefined
+          break
+        // adding/editing password
+        default:
+          password = link.password
+          break
+      }
+
+      return {
+        expireDate,
+        password,
+        permissions: link.permissions,
+        name: link.name,
+        ...(this.currentStorageId && {
+          spaceRef: `${this.currentStorageId}${this.highlightedFile.path}`
+        })
+      }
+    },
+
+    async createLink({ link, showError }) {
+      const params = this.getParamsForLink(link)
+
+      await this.addLink({
+        path: this.highlightedFile.path,
+        client: this.$client,
+        $gettext: this.$gettext,
+        params,
+        storageId: this.currentStorageId
+      }).catch(showError)
+
+      this.currentView = VIEW_SHOW
+
+      this.showMessage({
+        title: this.$gettext('Link was created successfully')
       })
+    },
+
+    async updatePublicLink({ link, onSuccess = () => {}, onError = (e) => {} }) {
+      const params = this.getParamsForLink(link)
 
-      this.currentView = VIEW_EDIT
+      await this.updateLink({
+        id: link.id,
+        client: this.$client,
+        params
+      })
+        .then(onSuccess)
+        .catch((e) => {
+          onError(e)
+          console.error(e)
+          this.showMessage({
+            title: this.$gettext('Failed to update link'),
+            status: 'danger'
+          })
+        })
+
+      this.showMessage({
+        title: this.$gettext('Link was updated successfully')
+      })
+    },
+
+    deleteLinkConfirmation({ link }) {
+      const modal = {
+        variation: 'danger',
+        title: this.$gettext('Delete link'),
+        message: this.$gettext(
+          'Are you sure you want to delete this link? Recreating the same link again is not possible.'
+        ),
+        cancelText: this.$gettext('Cancel'),
+        confirmText: this.$gettext('Delete'),
+        onCancel: this.hideModal,
+        onConfirm: () =>
+          this.deleteLink({
+            client: this.$client,
+            share: link,
+            resource: this.highlightedFile
+          })
+      }
+      this.createModal(modal)
+    },
+
+    async deleteLink({ client, share, resource }) {
+      this.hideModal()
+      // removeLink currently fetches all shares from the backend in order to reload the shares indicators
+      // TODO: Check if to-removed link is last link share and only reload if it's the last link
+      await this.removeLink({ client, share, resource, storageId: this.currentStorageId })
+        .then(
```

Comment:
please avoid mixing of then and await.

Reply:
Wasn't it us two that came up with this? Glad to replace it with something else, any suggestions?